### PR TITLE
doc: refactor the overview topic in the sphinx project

### PR DESCRIPTION
**Pull request by eslesar-aws**
*Issue #, if available:*

*Description of changes:*
Refactored the overview.rst content in the sphinx doc project per the updated structure.
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/overview.rst [hunk 1]
```diff
@@ -8,13 +9,82 @@ SageMaker Python SDK provides several high-level abstractions for working with A
 - **Predictors**: Provide real-time inference and transformation using Python data-types against a SageMaker endpoint.
 - **Session**: Provides a collection of methods for working with SageMaker resources.
 
-``Estimator`` and ``Model`` implementations for MXNet, TensorFlow, Chainer, PyTorch, and Amazon ML algorithms are included.
+``Estimator`` and ``Model`` implementations for MXNet, TensorFlow, Chainer, PyTorch, scikit-Learn, Amazon SageMaker built-in algorithms, Reinforcement Learning,  are included.
```
Comment (mvsusp) [inline doc/overview.rst]:
```suggestion
``Estimator`` and ``Model`` implementations for MXNet, TensorFlow, Chainer, PyTorch, scikit-learn, Amazon SageMaker built-in algorithms, Reinforcement Learning,  are included.
```

File: doc/overview.rst [hunk 2]
```diff
@@ -8,13 +9,82 @@ SageMaker Python SDK provides several high-level abstractions for working with A
...
+1. Prepare a training script
+2. Create an estimator
+3. Call the `fit` method of the estimator
+
+After you train a model, you can save it, and then serve the model as an endpoint to get real-time inferences or get inferences for an entire dataset by using batch transform.
+
+Prepare a Training script
+=========================
+
+Your training script must be a Python 2.7 or 3.5 compatible source file.
```
Comment (mvsusp) [inline doc/overview.rst]:
```suggestion
Your training script must be a Python 2.7 or 3.6 compatible source file.
```

File: doc/overview.rst [hunk 3]
```diff
@@ -8,13 +9,82 @@ SageMaker Python SDK provides several high-level abstractions for working with A
...
 There's also an ``Estimator`` that runs SageMaker compatible custom Docker containers, enabling you to run your own ML algorithms by using the SageMaker Python SDK.
 
 .. contents::
+   :depth: 2
+
+*******************************************
+Train a Model with the SageMaker Python SDK
+*******************************************
+
+To train a model by using the SageMaker Pthon SDK, you:
```
Comment (laurenyu) [inline doc/overview.rst]: s/Pthon/Python

File: doc/overview.rst [hunk 4]
```diff
@@ -8,13 +9,82 @@ SageMaker Python SDK provides several high-level abstractions for working with A
...
+
+1. Prepare a training script
+2. Create an estimator
+3. Call the `fit` method of the estimator
```
Comment (laurenyu) [inline doc/overview.rst]: double backticks (it's rst)

File: doc/overview.rst [hunk 5]
```diff
@@ -162,112 +261,24 @@ Here is an example of how to define metrics:
...
+You can create an instance of the ``Estimator`` class with desired Docker image and use it as described in previous sections.
 
+Please refer to the full example in the examples repo:
 
-For detailed examples of running Docker in local mode, see:
+::
 
-- `TensorFlow local mode example notebook <https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/tensorflow_distributed_mnist/tensorflow_local_mode_mnist.ipynb>`__.
-- `MXNet local mode example notebook <https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/mxnet_gluon_mnist/mnist_with_gluon_local_mode.ipynb>`__.
+    git clone https://github.com/awslabs/amazon-sagemaker-examples.git
```
Comment (laurenyu) [inline doc/overview.rst]: I wonder if it might be better to rephrase these to instruct how to find them in a SageMaker Notebook Instance instead

Comment (eslesar-aws) [reply doc/overview.rst]: How about adding that? I think that there is a category of customers that wants to be able to look at a notebook without the overhead of starting a notebook instance.

Comment (laurenyu) [reply doc/overview.rst]: fair enough

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4934537b-e623-4332-b314-4fc099a91ea1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4dc2d113-012b-469c-bc8b-85d9aaf97b60%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=96c2d245-2362-4920-ad94-444e380d31a3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f625b026-ce96-4568-965c-82e37f8e216f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a28c554a-6c4b-4461-b74b-0db9b00cbe0f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e09e1aed-bcad-46bf-8147-c2a13f0ed10a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://hno1ypgnog.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=31f65b1e-5b0a-4ff3-82c2-ac6e0a4b5c31%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=269e8e85-4e31-4b52-89a5-674a834ea88e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://rp2is7byfb.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f885b337-48aa-474e-b359-d7517a864073%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=665c0a7b-8f8e-4dbb-a98e-60b31e126600%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://mdk06ma93j.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3e386085-f38b-4561-aa2e-0f1a3b1d535b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**